lib/record_store/provider/oracle_cloud_dns.rb

```diff
@@ -0,0 +1,215 @@
+require 'oci'
+
+module RecordStore
+  class Provider::OracleCloudDNS < Provider
+    class Error < StandardError; end
+
+    class << self
+      def client
+        config = OCI::Config.new
+        config.user = secrets["user"]
+        config.fingerprint = secrets["fingerprint"]
+        config.key_content = secrets["key_content"]
+        config.tenancy = secrets["tenancy"]
+        config.region = secrets["region"]
```

Comment:
Probably wants to add [a call to `config.validate`](https://github.com/oracle/oci-ruby-sdk/blob/master/lib/oci/config.rb#L108-L124) after setting the parameters ? So we fail fast when creating the provider.